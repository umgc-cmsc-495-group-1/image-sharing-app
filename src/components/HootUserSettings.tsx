import React from 'react';
import { Typography } from '@mui/material';
import { useParams } from 'react-router-dom';

function HootUser() {
    const { userId  } = useParams();
   return(
    <Typography>
       This is user {userId} settings
    </Typography>
   ); 
}

export default HootUser;
