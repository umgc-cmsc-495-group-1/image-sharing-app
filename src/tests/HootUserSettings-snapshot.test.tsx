import React from 'react';
import "@testing-library/react/dont-cleanup-after-each";
import '@testing-library/jest-dom';
import { 
  MemoryRouter,
  Routes,
  Route
} from 'react-router-dom';
import HootUserSettings from '../components/HootUserSettings';
import { create } from 'react-test-renderer';

function renderWithMemoryRouter(component: JSX.Element) {
  return create(
    <MemoryRouter initialEntries={['/user/settings']}>
      <Routes>
        <Route path='/user/settings' element={component} />
      </Routes>
    </MemoryRouter>
  );
}

it('Renders correctly',() => {
  const renderer = renderWithMemoryRouter(<HootUserSettings/>);
  expect(renderer.toJSON()).toMatchSnapshot();
});