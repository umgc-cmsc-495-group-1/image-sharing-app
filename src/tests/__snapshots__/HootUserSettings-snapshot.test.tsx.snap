// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<ul
  className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
>
  <a
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    disabled={false}
    href="/about/terms-of-service"
    onClick={[Function]}
    role="terms-of-service"
  >
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Terms of Service
      </span>
    </div>
  </a>
  <a
    className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    disabled={false}
    href="/about/privacy"
    onClick={[Function]}
    role="privacy-policy"
  >
    <div
      className="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
      >
        Privacy Policy
      </span>
    </div>
  </a>
</ul>
`;
